dockerfiles/che/rhel.Dockerfile

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2018-2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.2-345
+USER root
+ENV CHE_HOME=/home/user/codeready
+ENV JAVA_HOME=/usr/lib/jvm/jre
+RUN microdnf install java-11-openjdk-headless tar gzip shadow-utils findutils && \
+    microdnf update -y gnutls && \
+    microdnf -y clean all && rm -rf /var/cache/yum && echo "Installed Packages" && rpm -qa | sort -V && echo "End Of Installed Packages" && \
+    adduser -G root user && mkdir -p /home/user/codeready && \
+    # fix certs & dir permissions - see file references in entrypoint.sh
+    for d in /home/user ${JAVA_HOME}/lib/security; do \
+      mkdir -p ${d}; cp /etc/pki/ca-trust/extracted/java/cacerts ${d}/cacerts && chmod 644 ${d}/cacerts; \
+    done
+
+COPY entrypoint.sh /entrypoint.sh
+ENTRYPOINT ["/entrypoint.sh"]
+
+# NOTE: if built in Brew, use get-sources-jenkins.sh to pull latest
+# OR, if you intend to build the Che Server tarball locally, 
+# see https://github.com/redhat-developer/codeready-workspaces-productization/blob/master/devdoc/building/building-crw.adoc#make-changes-to-crw-and-re-deploy-to-minishift
+# then copy /home/${USER}/projects/codeready-workspaces/assembly/codeready-workspaces-assembly-main/target/codeready-workspaces-assembly-main.tar.gz into this folder
+COPY assembly/codeready-workspaces-assembly-main/target/codeready-workspaces-assembly-main.tar.gz /tmp/codeready-workspaces-assembly-main.tar.gz
```

Comment:
That is something that does not exist in upstream. I suggest having one more Dockerfile in https://github.com/redhat-developer/codeready-workspaces-productization that overrides this content as an additional layer.

Reply:
So you're suggesting that I should have a `COPY` line that does something with locally built `che` bits, and when reusing it downstream, I'd transform it via sed to `COPY` in the CRW bits instead? 

Seems like we're making things more complicated than they need to be, but you're right, the crw tarball isn't in the upstream repo.

Reply:
True, in upstream it's just:
```
ADD eclipse-che /home/user/eclipse-che
```
https://github.com/eclipse/che/blob/master/dockerfiles/che/Dockerfile#L29

But in https://github.com/redhat-developer/codeready-workspaces-productization we have only process docs. We could transform an upstream rhel.Dockerfile into downstream https://github.com/redhat-developer/codeready-workspaces/blob/master/Dockerfile via https://github.com/redhat-developer/codeready-workspaces/pull/325/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R317-R322 by adding some new rules (or moving the whole thing to use @sleshchenko 's templated approach, but that requires having references to Brew in upstream)